les/handler.go

```diff
@@ -698,6 +698,12 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 						p.Log().Warn("Failed to retrieve header for code", "block", *number, "hash", request.BHash)
 						continue
 					}
+					// Refuse to search stale state data in the database since looking for
+					// a non-exist key is kind of expensive.
+					if !pm.server.archiveMode && header.Number.Uint64()+core.TriesInMemory <= pm.blockchain.CurrentHeader().Number.Uint64() {
```

Comment:
Shouldn't we use CurrentBlock here (and below)?

Reply:
Also, could we perhaps separate out `pm.blockchain.CurrentBlock().NumberU64()` into a separate variable and reuse that inside the log too. One less data access + code becomes cleaner.

Reply:
The reason I use `CurrentHeader` here is: now we mix all the stuff together, so that the `blockchain` is shared by both server and client. 
While the `lightChain` of client doesn't have `CurrentBlock` function.
In the refactored version(server and client separation), we can do it easily. But now since we need do a bit more.

Reply:
Sure